### PR TITLE
NEW: Add EventsAPI

### PR DESCRIPTION
**Pull request by apj**

### CODE REVIEW

File: uncompiled/lib/api/events.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    this.app = app;
+  }
+
+  getEvents(queryOptions) {
+    let query = this._translateQueryOptions(queryOptions);
+
+    return got(`${this.app.api.url}/events`, {
+      method: "GET",
+      json: true,
+      query: query
```
Comment (lukehorvat) [inline uncompiled/lib/api/events.js]: If the value is the same name as the key, you can omit it.

```js
method: "GET",
json: true,
query
```

**Comment by apj** [comment]
@lukehorvat Thanks, updated.

File: uncompiled/lib/api/events.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                                   .utc()
+                                   .format();
+    }
+
+    // Event type (e.g. 'DeviceEvent', 'TaskScheduleEvent')
+    if (queryOptions.type) {
+      let types = null;
+
+      // Make it an array
+      if (_.isArray(queryOptions.type)) {
```
Comment (lukehorvat) [inline uncompiled/lib/api/events.js]: More idiomatic way imo:

```
[].concat(queryOptions.type)
```

**Comment by apj** [comment]
@lukehorvat Agree. Didn't realise `contact` was as flexible with what it takes :eggplant:

**Comment by lukehorvat** [comment]
@apj LGTM :ok_hand:

**Comment by lukehorvat** [comment]
On second thought, it might be good to remove the lodash dep, since it's now only being used for `isEmpty`.

**Comment by apj** [comment]
Test.

**Comment by apj** [comment]
:eggplant:

**Comment by apj** [comment]
@lukehorvat Donezo.

**Pull request merged**